src/osd/OSDMap.cc

```diff
@@ -4281,9 +4292,6 @@ bool OSDMap::try_pg_upmap(
   if (rule < 0)
     return false;
 
-  // get original mapping
-  _pg_to_raw_osds(*pool, pg, orig, NULL);
-
```

Comment:
This doesn't look right.  The caller is still passing in an empty orig vector<int>, which means the loop below isn't doing anything

Reply:
I introduce an extra `pg_to_raw_upmap` method to get the in-time remapping results since we are now being able to modify an existing `pg_upmap_items`. See comment below.